### PR TITLE
collection-scripts/gather: drop MCO resources

### PR DESCRIPTION
**Pull request by runcom**
The MCO relatedObjects field grabs those

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

**Comment by sferich888** [comment]
Are these not part of some other resource? Where you can add them as a relatedObject? 

If not, /lgtm.

**Comment by runcom** [comment]
> Are these not part of some other resource? Where you can add them as a relatedObject?

we can't do that unfortunately as, like machineconfigs, those are user provided and cannot predict their names - the MCO related objects are static in naming and we can always grab them

**Comment by enxebre** [comment]
@sferich888 
with the PR in its current state where would this logs go in the must gather folder tree?

>we can't do that unfortunately as, like machineconfigs, those are user provided and cannot predict their names - the MCO related objects are static in naming and we can always grab them

@runcom You can grab all resources of a type as relatedResources, e.g we do this for machines, machineSets and CSRs see -> https://github.com/openshift/machine-api-operator/blob/35a3cd0b1d470d024c65d6237b17251b988fbc98/pkg/operator/status.go#L168-L177
https://github.com/openshift/cluster-machine-approver/blob/4e5ad9c9430c6b6a383dcba86107dff3e64bd686/status.go#L37-L48

**Comment by runcom** [comment]
> @sferich888
> with the PR in its current state where would this logs go in the must gather folder tree?

they'll go under machineconfiguration.openshift.io under cluster-scoped resources afact

> @runcom You can grab all resources of a type as relatedResources, e.g we do this for machines, machineSets and CSRs see -> https://github.com/openshift/machine-api-operator/blob/35a3cd0b1d470d024c65d6237b17251b988fbc98/pkg/operator/status.go#L168-L177
> https://github.com/openshift/cluster-machine-approver/blob/4e5ad9c9430c6b6a383dcba86107dff3e64bd686/status.go#L37-L48

@enxebre where is it documented that providing an empty Name gives you all the resources? also, are you getting all resources one by one? or just the output of e.g. oc get machines?

**Comment by runcom** [comment]
> @enxebre where is it documented that providing an empty Name gives you all the resources? also, are you getting all resources one by one? or just the output of e.g. oc get machines?

uhm, seems like you're getting everything indeed (e.g. oc describe <resource> -o yaml):

/home/amurdaca/must-gather.local.1975743706142649508/quay-io-openshift-release-dev-ocp-v4-0-art-dev-sha256-9734e6d304b88b243df5a1720570bf63ab684b2317755e0ed00de0aa32b05ce8/namespaces/openshift-machine-api/machine.openshift.io/machines

Wondering if that's true for cluster-scoped resources

**Comment by runcom** [comment]
holding pending tests on https://github.com/openshift/machine-config-operator/pull/1779

that might requires that we drop the collection here altogether as we'll do that directly with the MCO/CVO

/hold

File: collection-scripts/gather [hunk 1]
```diff
@@ -18,7 +18,7 @@ resources+=(clusteroperators)
 resources+=(certificatesigningrequests)
 
 # Machine/Node Resources
-resources+=(nodes machineconfigs machineconfigpools)
+resources+=(nodes machineconfigs machineconfigpools kubeletconfigs containerruntimeconfigs)
```
Comment (runcom) [inline collection-scripts/gather:21]: if https://github.com/openshift/machine-config-operator/pull/1779 works, I think we'll drop everything here except `nodes`

**Comment by runcom** [comment]
> holding pending tests on [openshift/machine-config-operator#1779](https://github.com/openshift/machine-config-operator/pull/1779)

doesn't seem to be working https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_machine-config-operator/1779/pull-ci-openshift-machine-config-operator-master-e2e-aws/7545/artifacts/e2e-aws/gather-must-gather/ @enxebre uhm

File: collection-scripts/gather (hunk 1 above)
Comment (runcom) [reply collection-scripts/gather]: pushed a fix!

**Comment by runcom** [comment]
can we get this in @sferich888 

/retest

**Pull request merged**